src/_guides/language/effective-dart/style.md

```diff
@@ -337,6 +337,21 @@ for you.
 [dartfmt]: https://github.com/dart-lang/dart_style
 
 
+### CONSIDER changing your code to make it more formatter-friendly.
+
+The formatter does the best it can with whatever code you throw at it, but it
+can't work miracles. If your code has particularly long identifiers, deeply
+nested expressions, a mixture of different kinds of operators, etc. the
+formatted output may still be hard to read.
+
+When that happens, reorganize or simplify your code. Consider shortening a local
+variable name or hoisting out an expression into a new local variable. In other
+words, the exact kinds of modifications you'd make if you were hand-formatting
```

Comment:
Let's make this a sentence and reword a bit for easier reading. Maybe:

In other words, make the same kinds of modifications that you'd make if you were formatting the code by hand and trying to make it more readable.

Reply:
Done.